documentation/writing_tests.md

```diff
@@ -14,6 +20,139 @@
 
 ## Recommended tools
 
+Writing tests for JavaScript and TypeScript requires testing tools, such as a test framework, an assertion library and a way to bundle and run these tests on various environments. The JavaScript community has many overlapping tools that one could pick to fulfill any of these tasks. To ensure that the testing experience across the Azure SDK for JS/TS is consistent and reliable,
+we've picked the following external testing tools:
+
+- [Mocha](https://www.npmjs.com/package/mocha), which offers us a well known and stable test framework for both NodeJS and browser tests.
+- [Chai](https://www.npmjs.com/package/chai), a well known assertion library for Node and the browser.
+- [Rollup](https://www.npmjs.com/package/rollup), to bundle JavaScript for different environments, which helps us write a single TypeScript source and trust it will compile correctly for Node and the browser.
```

Comment:
maybe some note that Rollup is good at creating efficient and small bundles for the browser out of ES6 modules?

Reply:
I just pushed a whole section on Rollup! Let me know if you like it.